slack/web/classes/objects.py

```diff
@@ -213,83 +230,86 @@ class ConfirmObject(JsonObject):
     confirm_max_length = 30
     deny_max_length = 30
 
+    @classmethod
+    def parse(cls, confirm: Union["ConfirmObject", dict]):
+        if confirm:
+            if isinstance(confirm, ConfirmObject):
+                return confirm
+            elif isinstance(confirm, dict):
+                return ConfirmObject(**confirm)
+            else:
+                # TODO: warning
+                return None
+
     def __init__(
         self,
         *,
-        title: str,
-        text: Union[TextObject, str],
-        confirm: str = "Yes",
-        deny: str = "No",
+        title: Union[str, dict, PlainTextObject],
+        text: Union[str, dict, TextObject],
+        confirm: Union[str, dict, PlainTextObject] = "Yes",
+        deny: Union[str, dict, PlainTextObject] = "No",
     ):
         """
         An object that defines a dialog that provides a confirmation step to any
         interactive element. This dialog will ask the user to confirm their action by
         offering a confirm and deny button.
-
-        https://api.slack.com/reference/messaging/composition-objects#confirm
-
-        Args:
-            title: A string that defines the dialog's title. Cannot exceed 100
-                characters.
-            text: A string or TextObject that defines the explanatory text
-                that appears in the confirm dialog. Cannot exceed 300 characters.
-            confirm: A string to define the text on the button that confirms the
-                action. Cannot exceed 30 characters.
-            deny: A string to define the text on the button that cancels the
-                action. Cannot exceed 30 characters.
+        https://api.slack.com/reference/block-kit/composition-objects#confirm
         """
-        self.title = title
-        self.text = text
-        self.confirm = confirm
-        self.deny = deny
+        self.title = TextObject.parse(title, default_type=PlainTextObject.type)
+        self.text = TextObject.parse(text, default_type=MarkdownTextObject.type)
```

Comment:
I just noticed here is a breaking change. The outcome of `to_dict()` is compatible so that I don't think this has a major impact. That said, this is not compatible for the users accessing `ConfirmObject#text` after initializing the object. The type used to be `str` but is `TextObject` now. 

TODO: I will change the code to have `_text: TextObject` and modify `to_dict` override.

Reply:
Fixed this issue with the latest commit.